orttraining/orttraining/training_ops/cuda/communication/recv.cc

```diff
@@ -72,15 +73,15 @@ void Recv::ReceiveData(
   // count waiting time before setting up the actual communication.
   recvRange.Begin();
 #endif
-  buffer = AllocateBufferOnCPUPinned<char>(static_cast<size_t>(aggregated_aligned_tensor_bytes));
+  buffer = GetScratchBuffer<char>(aggregated_aligned_tensor_bytes);
+
   CommInfo_t info_data{buffer.get(),
                        static_cast<int>(aggregated_aligned_tensor_bytes),
                        src,
                        static_cast<int>(tag_)};
 
-  MPI_CHECK(MPI_Recv(
-      info_data.buffer, info_data.size, MPI_CHAR,
-      info_data.rank, info_data.tag, MPI_COMM_WORLD, MPI_STATUS_IGNORE));
```

Comment:
Could we preserve the MPI_Send/Recv path? It might be useful for debugging

Reply:
I can make them comments but not actual code. What do you think?

---
In reply to: [483804058](https://github.com/microsoft/onnxruntime/pull/5054#discussion_r483804058) [](ancestors = 483804058)